st2common/st2common/transport/bootstrap_utils.py

```diff
@@ -69,3 +72,12 @@ def wrapped_register_exchanges(connection, channel):
                                       retry_wrapper=retry_wrapper)
 
         retry_wrapper.run(connection=conn, wrapped_callback=wrapped_register_exchanges)
+
+
+def register_exchanges_with_retry():
+    retrying_obj = retrying.Retrying(
+        retry_on_exception=socket.error,
+        wait_fixed=cfg.CONF.messaging.connection_retry_wait,
+        stop_max_attempt_number=cfg.CONF.messaging.connection_retries
```

Comment:
If this property is only used for retries in register_exchanges why is it called connection_retries? The issue with a broadly named property like this is that it is bound to confuse someone as to why this is getting st2 to retry the specified no of times.

Perhaps there is value in wiring these properties up to https://github.com/StackStorm/st2/blob/master/st2common/st2common/transport/connection_retry_wrapper.py#L30. The semantics there are slightly different but worth looking into to bring together to mean the same thing.